tests/test_util/aiohttp.py

```diff
@@ -74,13 +76,24 @@ class AiohttpClientMockResponse:
     def __init__(self, method, url, status, response):
         """Initialize a fake response."""
         self.method = method
-        self.url = url
+        self._url = url
+        try:
+            self._url_parts = urlparse(url.lower())
+        except AttributeError:
+            self._url_parts = None
         self.status = status
         self.response = response
 
     def match_request(self, method, url):
         """Test if response answers request."""
-        return method == self.method and url == self.url
+        if method.lower() != self.method.lower():
+            return False
+
+        # Reuse Mock request's match function
+        # pylint: disable=protected-access, attribute-defined-outside-init
+        from requests_mock.adapter import _Matcher
```

Comment:
Ok, will revert to blatant plagiarism :-P

Makes sense if you want to contribute it back to aiohttp